### PR TITLE
Remove duplicate code

### PR DESCRIPTION
**Pull request by satybald**
While working on #1181, found code that violates DRY principle and needs improvement. Added tests to the refactored code.

### CODE REVIEW

File: ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java [hunk 1]
```diff
@@ -639,4 +651,19 @@ public void shouldSetShowDescriptionsForShowQueriesDescriptions() {
     ListQueries listQueries = (ListQueries)statement;
     Assert.assertThat(listQueries.getShowExtended(), is(true));
   }
+  
+    private void assertQuery(Statement statement) {
```
Comment (big-andy-coates) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: nit: indentation

Comment (big-andy-coates) [reply ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: Test would also read better if this method was `assertQuerySucceeds("SELECT * FROM BLAH")`.

File: ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java [hunk 2]
```diff
@@ -639,4 +651,19 @@ public void shouldSetShowDescriptionsForShowQueriesDescriptions() {
...
+    assertTrue(statement instanceof Query);
+  }
+
+  private void assertFailedQuery(String sql, String exceptionMessage) {
```
Comment (big-andy-coates) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: Code would read better if this was named `assertQueryFails`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java [hunk 3]
```diff
@@ -639,4 +651,19 @@ public void shouldSetShowDescriptionsForShowQueriesDescriptions() {
...
+    try {
+      KSQL_PARSER.buildAst(sql, metaStore).get(0);
+      fail(format("Expected query: %s to fail with message: %s", sql, exceptionMessage));
+    } catch (RuntimeException exp) {
+      if(!exp.getMessage().equals(exceptionMessage)) {
```
Comment (big-andy-coates) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: Consider checking exception type

File: ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java [hunk 4]
```diff
@@ -639,4 +651,21 @@ public void shouldSetShowDescriptionsForShowQueriesDescriptions() {
     ListQueries listQueries = (ListQueries)statement;
     Assert.assertThat(listQueries.getShowExtended(), is(true));
   }
+  
+  private void assertQuerySucceeds(Statement statement) {
+    assertTrue(statement instanceof Query);
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: nit: `assertThat(statement, instanceOf(Query.class))`

File: ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java [hunk 5]
```diff
@@ -445,7 +462,7 @@ public void testShouldFailIfWrongKeyword() {
     try {
       String simpleQuery = "SELLECT col0, col2, col3 FROM test1 WHERE col0 > 100;";
       Statement statement = KSQL_PARSER.buildAst(simpleQuery, metaStore).get(0);
-      Assert.fail();
+      fail();
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: I know this was here before, but would be useful to have a reason why it failed

File: ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java [hunk 6]
```diff
@@ -639,4 +651,21 @@ public void shouldSetShowDescriptionsForShowQueriesDescriptions() {
...
+
+  private void assertQueryFails(String sql, String exceptionMessage) {
+    try {
+      KSQL_PARSER.buildAst(sql, metaStore).get(0);
+      fail(format("Expected query: %s to fail with message: %s", sql, exceptionMessage));
+    } catch (ParseFailedException exp) {
+      if(!exp.getMessage().equals(exceptionMessage)) {
+        fail(format("Expected exception message to match %s for query: %s", exceptionMessage, sql));
+      }
+    } catch (RuntimeException exp) {
```
Comment (dguy) [inline ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: Don't need this

Comment (satybald) [reply ksql-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java]: could you please elaborate why it's not needed?

**Comment by satybald** [comment]
thanks @dguy. I've applied your feedback.

**Comment by satybald** [comment]
thanks, @dguy for the hint. would you mind to review PR once again?

**Comment by dguy** [comment]
@satybald can you please merge the latest from master in? If the build succeeds i'll commit it. Thanks!

**Comment by satybald** [comment]
@dguy, sure did so. jenkins finally build successfully.

**Comment by dguy** [comment]
Thanks @satybald, merged to master

**Pull request merged**